src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -1504,6 +1504,16 @@ def index(self) -> Union["pd.Index", native_pd.MultiIndex]:
         else:
             return pd.Index(self)
 
+    def pandas_index(self) -> native_pd.Index:
```

Comment:
try to reuse `to_pandas`, you can add an argument like `index_only = True`.

Reply:
or call `self._modin_frame.index_columns_index` inside `to_pandas`.

Reply:
> or call `self._modin_frame.index_columns_index` inside `to_pandas`.

I would like to do this, is it ok for me to access modin frame?

Reply:
Yes, you can access `_modin_frame` in `snowflake_query_compiler.py`

Reply:
Any reason we cannot use `to_pandas` to implement `index_columns_index`? It will make our code clean and concise.